script/tool/test/gradle_check_command_test.dart

```diff
@@ -789,6 +869,72 @@ dependencies {
         isNot(contains(GradleCheckCommand.exampleRootGradleArtifactHubString)),
       );
     });
+
+    test('prints error for declarative method of applying gradle plugins', () async {
+      const String packageName = 'a_package';
+      final RepositoryPackage package =
+      createFakePackage('a_package', packagesDir);
+      writeFakePluginBuildGradle(package, includeLanguageVersion: true);
+      writeFakeManifest(package);
+      final RepositoryPackage example = package.getExamples().first;
+      writeFakeExampleBuildGradleGP(example,
+          pluginName: packageName,
+          // ignore: avoid_redundant_argument_values
```

Comment:
If we always want to have the tests be explicit about all of their arguments, then let's make the test helper arguments be `required` instead of having to annotate all of the calls.

Reply:
I dont see the version of the code where includeBuildArtifactHub and includeSettingsDocumentationArtifactHub are required without defaults  and the `// ignore: avoid_redundant_argument_values` are removed. Did you need to push a change?

Reply:
I added `required` and removed the annotations for `writeFakeExampleBuildGradleGP` but not for `writeFakeExampleBuildGradles`. Let me update that